### PR TITLE
Optional filter

### PR DESCRIPTION
**Pull request by Atlanex**
# Description of the PR
Filter for optional assignments. This way past weeks with non-completed optional assignments can be hidden

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [x] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](/TESTING.md) or other relevant documentation?

- [x] Yes
- [ ] Not yet. I will do it next.
- [ ] Not relevant

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java [hunk 1]
```diff
@@ -123,7 +123,12 @@ public int getMaxSubmissions() {
   }
 
   public boolean isCompleted() {
-    return userPoints == maxPoints && submissionResults.size() > 0;
+    // Optional assignments are never completed, since they can be filtered separately
```
Comment (nikke234) [inline src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java:126]: `...and we can't tell from the points whether the submission was correct or not`

File: src/test/java/fi/aalto/cs/apluscourses/model/ExerciseTest.java [hunk 2]
```diff
@@ -175,4 +175,15 @@ public void testIsCompleted() {
...
   }
+
+  @Test
+  public void testIsOptional() {
+    Exercise optional = new Exercise(1, "optional", "http://localhost:1111", 0, 0, 0);
+    assertTrue("Assignment is optional",
+            optional.isOptional());
+
+    Exercise notOptional = new Exercise(2, "notOptional", "http://localhost:1111", 0, 5, 10);
+    assertFalse("Assingment isn't optional",
```
Comment (nikke234) [inline src/test/java/fi/aalto/cs/apluscourses/model/ExerciseTest.java:186]: Typo: Assingment -> Assignment :smile:

Comment (Atlanex) [reply src/test/java/fi/aalto/cs/apluscourses/model/ExerciseTest.java:186]: Whoops

**Review by OlliKiljunen — APPROVED** [review]
Code + manual testing instructions look perfect and I also understood this is a feature we wanted to have so an easy approve 😎👍 Thanks!

**Review by nikke234 — APPROVED** [review]
Great stuff once again, and props for updating the manual testing documentation :+1:

**Pull request merged**